lib/externalGradingSocket.sql

```diff
@@ -20,16 +20,11 @@ SELECT
     grading_job_status(gj.id) AS grading_job_status,
     s.variant_id AS variant_id
 FROM
-    grading_jobs AS gj
+    grading_jobs AS gj_orig
+    JOIN grading_jobs AS gj ON (gj.submission_id = gj_orig.submission_id)
     JOIN submissions AS s ON (s.id = gj.submission_id)
 WHERE
-    gj.id = (
-        SELECT MAX(gj2.id)
-        FROM grading_jobs AS gj2
-        WHERE gj2.submission_id = (
-            SELECT s2.id
-            FROM submissions AS s2
-            RIGHT JOIN grading_jobs AS gj3 ON (gj3.submission_id = s2.id)
-            WHERE gj3.id = $grading_job_id
-        )
-    );
+    gj_orig.id = $grading_job_id
+    AND gj.grading_method != 'Manual'
+ORDER BY gj.date DESC, gj.id DESC
+LIMIT 1;
```

Comment:
It looks like this query is invoked with `queryOneRow` - is there a risk of ending up with zero rows? Given that the function that uses this query should only be called in response to an external grading job completing, I think we're safe, I just want to double check.

Reply:
I see this query is called with a known grading job number, so presumably it should return the original grading job itself if no other grading job exists. I checked all references to this function, and all of them seem to work with the assumption that the job exists.

Reply:
Good point; I'm satisfied that this is good to go!